File: prime/prime.c
#include <stdio.h>
#include <math.h>
#include <stdlib.h>

#define TRUE 1
#define MAX 9999

int is_prime(int n){
    int i, sqr;
    sqr = (int)sqrt(n);
    for(int i = 2; i <= sqr; i++){
        if(n%i==0){
            return !TRUE;
        }
    }
    return TRUE;
}

int get_prime(int n){
    int* iptr;
    int i,j;
    
    iptr = (int*)calloc(MAX, sizeof(int));
    if(iptr == NULL){
        puts("\n Memory allocation error!");
        exit(1);
    }
    
    for(i=2; i<MAX; i++){
        if(iptr[i] == 1)
            continue;
        j=i;
        while((j+=i) <= MAX)
            iptr[j] = 1;
    }

    for (i=2; i<=MAX; i++){
        if(iptr[i] == 0)
            printf("\t%6d", i);
    }
}

//This iptr makes MAX num's 0
//if j+i which will be i+i must be not prime (can be divided by i)

void main(void){
    int x;
    scanf("put number %d\n\n", &x);
    printf("Result 1 is %d\n", is_prime(x));
    printf("Result 2 is\n");
    get_prime(x);
}

File: prime/prime.py
import math

def is_prime(n):
    sqr = int(math.sqrt(n))
    for i in range(2,sqr+1):
        if(n%i == 0):
            return False        
    return True

def get_prime(n):
    temp = []

    for i in range(n):
        temp.append(0)

    for i in range(2, n):
        if(temp[i] == 1):
            continue
        j=i
        while(j+i <= len(temp)-1):
            j += i
            temp[j] = 1

    for i,(X) in enumerate(temp):
        if(X == 0):
            print(i, end=" ")
            
# have to fix

x = int(input("put number "))
print("Result 1 is ", is_prime(x))
print("Result 2 is");
get_prime(x)